### PR TITLE
Add avro docs

### PR DESCRIPTION
**Pull request by mwylde**

### CODE REVIEW

File: connectors/formats.mdx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+## Avro
+
+[Avro](https://avro.apache.org/) is a binary data format that is commonly used for
+data applications. Avro is a schema-based format, and requires readers and writers to
+have access to the schema in order to read and write data.
+
+Avro is well-supported by the Kafka ecosystem and the
+[Confluent Schema Registry](https://docs.confluent.io/platform/current/schema-registry/index.html),
+is a popular choice for storing and serving Avro schemas. Arroyo has deep support for the Schema Registry.
```
Comment (jacksonrnewhouse) [inline connectors/formats.mdx:51]: I think this first sentence should be reworded or broken up into two.

File: connectors/formats.mdx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+The following options are supported:
+
+| Option | Description | Default |
+| ------ | ----------- | ------- |
+| `format` | The format of the data. Must be `avro` | `avro` |
+| `avro.confluent_schema_registry` | Set to true if data was produced by (or will be consumed via) a [Confluent Schema Registry](https://docs.confluent.io/platform/current/schema-registry/index.html) connected source | `false` |
+| `avro.raw_datums` | Set to true to serialize and deserialize as raw Avro datums instead of complete Avro records | `false` |
+| `avro.into_unstructured_json` | Convert the avro record to JSON | `false` |
+
+Avro data can be serialized/deserilaized in two ways:
```
Comment (jacksonrnewhouse) [inline connectors/formats.mdx:62]: spelling: deserialized

File: connectors/formats.mdx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+* As a complete Avro document, which includes the schema and metadata
+* As a raw Avro datum, which is just the data itself
+
+In the former mode, applications reading from the data will need to have access to the exact schema
```
Comment (jacksonrnewhouse) [inline connectors/formats.mdx:67]: The "former mode"/"latter mode" don't seem to match the order you list the serialization options.

File: connectors/formats.mdx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+When using Avro with a Kafka sink, Arroyo will automatically register the schema with the Schema Registry so
+long as the `avro.confluent_schema_registry` option is set to `true`. This allows the schema to be used by
+other applications that read from the same topic.
+
+## Raw string
+
+Raw string data is treated as a single column named `value` with type `TEXT`,
+and can be operated on using
+[SQL string functions](/sql/scalar-functions#string-functions) or UDFs.
```
Comment (jacksonrnewhouse) [inline connectors/formats.mdx:93]: Link to the JSON UDFs or the UDF page in general?

File: connectors/overview.mdx [hunk 5]
```diff
@@ -51,66 +51,11 @@ to configure them in Arroyo.
 
 ## Connection formats
 
-Some connectors support arbitrary data formats, while others have a fixed format defined
-by the connector. For example, Kafka stores data as a sequence of bytes, which can be
-interpreted in many different ways. When creating a connection it's necessary to specify
-the format of the data so that Arroyo can serialize or deserialize it.
+Arroyo supports a number of different data formats for reading and writing data, including
+JSON, Avro, Parquet, and raw strings. The format is configured via the `format` option
```
Comment (jacksonrnewhouse) [inline connectors/overview.mdx:55]: link to format page here?

**Review by jacksonrnewhouse — CHANGES_REQUESTED** [review]
Just a few things to address, looks good overall.

File: connectors/overview.mdx (hunk 5 above)
Comment (mwylde) [reply connectors/overview.mdx:55]: It's linked at the bottom of the section.

File: connectors/formats.mdx [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+## Avro
+
+[Avro](https://avro.apache.org/) is a binary data format that is commonly used for
+data applications. Avro is a schema-based format, and requires readers and writers to
+have access to the schema in order to read and write data.
+
+Avro is well-supported by the Kafka ecosystem where the
+[Confluent Schema Registry](https://docs.confluent.io/platform/current/schema-registry/index.html),
+is a popular choice for storing and serving Avro schemas. Aroryo is able to read and write
```
Comment (jbeisen) [inline connectors/formats.mdx:51]:
```suggestion
is a popular choice for storing and serving Avro schemas. Arroyo is able to read and write
```

**Pull request merged**